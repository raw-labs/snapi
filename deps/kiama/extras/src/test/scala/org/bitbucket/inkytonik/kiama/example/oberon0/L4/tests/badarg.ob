MODULE badarg;

TYPE
    arr1 = ARRAY 5 OF BOOLEAN;
    arr2 = ARRAY 5 OF BOOLEAN;
    rec1 = RECORD x : INTEGER END;
    rec2 = RECORD x : INTEGER END;

VAR
    arr : arr2;
    rec : rec2;

PROCEDURE nonidparam (a : ARRAY 10 OF INTEGER; b : RECORD x : INTEGER END);

END nonidparam;

PROCEDURE nonvararrayparam (a : arr1; b : rec1);

END nonvararrayparam;

PROCEDURE One (VAR a : arr1; VAR b : rec1);

PROCEDURE Two (VAR c : arr2; VAR d : rec2);

BEGIN
END Two;

BEGIN
    Two (a, b)
END One;

BEGIN
    nonvararrayparam (arr, rec)
END badarg.
