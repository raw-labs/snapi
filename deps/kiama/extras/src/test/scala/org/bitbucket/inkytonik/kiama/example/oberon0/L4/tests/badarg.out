* _pp.ob
MODULE badarg;

    TYPE
        arr1 = ARRAY 5 OF BOOLEAN;
        arr2 = ARRAY 5 OF BOOLEAN;
        rec1 = RECORD x : INTEGER END;
        rec2 = RECORD x : INTEGER END;

    VAR
        arr : arr2;
        rec : rec2;

    PROCEDURE nonidparam (a : ARRAY 10 OF INTEGER; b : RECORD x : INTEGER END);
    
    END nonidparam;
    
    PROCEDURE nonvararrayparam (a : arr1; b : rec1);
    
    END nonvararrayparam;
    
    PROCEDURE One (VAR a : arr1; VAR b : rec1);
    
        PROCEDURE Two (VAR c : arr2; VAR d : rec2);
        
        BEGIN
            
        END Two;
    
    BEGIN
        Two (a, b)
    END One;

BEGIN
    nonvararrayparam (arr, rec)
END badarg.
[3]* errors
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:13:27:error: parameter type must be identifier
[3]PROCEDURE nonidparam (a : ARRAY 10 OF INTEGER; b : RECORD x : INTEGER END);
[3]                          ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:13:52:error: parameter type must be identifier
[3]PROCEDURE nonidparam (a : ARRAY 10 OF INTEGER; b : RECORD x : INTEGER END);
[3]                                                   ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:17:29:error: array parameter must be VAR
[3]PROCEDURE nonvararrayparam (a : arr1; b : rec1);
[3]                            ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:17:39:error: record parameter must be VAR
[3]PROCEDURE nonvararrayparam (a : arr1; b : rec1);
[3]                                      ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:29:10:error: got arr1, but expected arr2
[3]    Two (a, b)
[3]         ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:29:13:error: got rec1, but expected rec2
[3]    Two (a, b)
[3]            ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:33:23:error: got arr2, but expected arr1
[3]    nonvararrayparam (arr, rec)
[3]                      ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L4/tests/badarg.ob:33:28:error: got rec2, but expected rec1
[3]    nonvararrayparam (arr, rec)
[3]                           ^
